dc-commons-xml/src/main/java/de/digitalcollections/commons/xml/xpath/XPathMapper.java

```diff
@@ -265,6 +311,72 @@ private String resolveVariableContext(Locale language, String variableContext, M
     return result;
   }
 
+  /**
+   * Verification of the return type.
+   * <br/>
+   * <ul>
+   * <li>Multi language fields can return single and multi valued content, so the return types are
+   * <code>Map&lt;Locale,String&gt;</code> and <code>Map&lt;Locale,List&lt;String&gt;&gt;</code>, but
+   * multi valued content is only allowed on expressions, not in template evaluations.
+   * <li>For a single valued, non localized field, it must be <code>String</code>
+   * <li>For a multi valued, non localized field, it must be <code>List&lt;String&gt;</code>
+   * </ul>
+   * @param method the invoked method
+   * @param binding the XPathBinding
+   * @throws XPathMappingException if the validation did not pass
+   */
+  private void verifyReturnType(Method method, XPathBinding binding) throws XPathMappingException {
+    // If we insist in using the multiLanguage annotation, we must check the return type
+    if (binding.multiLanguage()) {
+      if (!method.getReturnType().isAssignableFrom(Map.class)) {
+        throw new XPathMappingException(
+            "Method return type must be " + LOCALIZED_SINGLEVALUE_RETURN_TYPE
+                + " if multiLanguage=true");
+      }
+      return;
+    }
+
+    // If we use a multiLanguage field, we ensure, that the return type is Map<Locale,String> or Map<Locale,List<String>>
+    if (method.getReturnType().isAssignableFrom(Map.class)) {
+      String genericReturnTypeName = method.getGenericReturnType().getTypeName();
+
+      if (!LOCALIZED_SINGLEVALUE_RETURN_TYPE.isSubtypeOf(method.getGenericReturnType())
```

Comment:
Ich würde `method.getGenericReturnType()` in eine lokale Variable auslagern, das wird hier ja in fast jedem Statement gebraucht.

Reply:
done